app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java

```diff
@@ -18,52 +35,309 @@
+        timeText.setTextSize(TypedValue.COMPLEX_UNIT_SP, TEXT_SIZE_14);
+        time.addView(timeText);
+        return timeText;
+    }
+
+    /**
+     * Initialize the time section of the dialog.
+     * @param time The layout to put the time and date
+     * @return The dateText
+     */
+    private TextView initializeTimeLayout(LinearLayout time) {
+        time.setOrientation(LinearLayout.HORIZONTAL);
+        time.setGravity(Gravity.CENTER_HORIZONTAL);
+        time.setWeightSum(2);
+        TextView dateText = new TextView(new ContextThemeWrapper(getContext(), R.style.HintStyle));
+        Date currentTime = Calendar.getInstance().getTime();
+        SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT, Locale.getDefault());
+        dateText.setText(dateFormat.format(currentTime));
+        dateText.setTextColor(Color.parseColor(PRIMARY_COLOR));
+        dateText.setTextSize(TypedValue.COMPLEX_UNIT_SP, TEXT_SIZE_12);
+        time.addView(dateText);
+        return dateText;
+    }
+
+    /**
+     * Initialize the spinner of the dialog.
+     * @param layout The layout to put the spinner
+     * @return The spinner
+     */
+    private Spinner initializeSpinner(LinearLayout layout) {
+        final ArrayAdapter<String> adp = new ArrayAdapter<>(Objects.requireNonNull(getContext()),
+                android.R.layout.simple_spinner_item, (List<String>) getPetsName());
+        Spinner sp = new Spinner(getContext());
+        sp.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,
+                ViewGroup.LayoutParams.WRAP_CONTENT));
+        sp.setAdapter(adp);
+        layout.addView(sp);
+        return sp;
+    }
+
+    /**
+     * Indicates is a string is in a correct time format.
+     * @param time The time string
+     * @return True if the string is in a correct time format, false otherwise
+     */
+    private boolean isValidTime(String time) {
+        boolean result = false;
+        if (time.length() == TIME_LENGTH && time.indexOf(':') == POS2 && time.lastIndexOf(':') == POS5) {
+            if (Character.isDigit(time.charAt(0)) && Character.isDigit(time.charAt(1))
+                    && Character.isDigit(time.charAt(POS3))) {
+                if (Character.isDigit(time.charAt(POS4)) && Character.isDigit(time.charAt(POS6))
+                        && Character.isDigit(time.charAt(POS7))) {
+                    result = true;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Select a pet for a new event.
+     * @param petName The name of the pet
+     */
+    private void getPet(String petName) {
+        ArrayList<Pet> pets = user.getPets();
+
+        for (Pet pet : pets) {
+            if (pet.getName().equals(petName)) {
+                selectedPet = pet;
+            }
+        }
+    }
+
+    /**
+     * Gets all the pets of the user.
+     * @return An array with the pets name
+     */
+    private ArrayList<?> getPetsName() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using implementation types like 'ArrayList'; use the interface instead](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)